pipeline-templates/job/deploy-grafana.yml

```diff
@@ -0,0 +1,199 @@
+parameters:
+  AksClusterName: 
+  AksResourceGroupName: 
+  AppGatewayName:
+  AppGatewayResourceGroup: 
+  AzureServiceConnection: 
+  Environment: 
+  GrafanaBaseUri: 
+  GrafanaAdminPassword: 
+  GrafanaAdminUsername:
+  HelmClientVersion: 
+  HelmChartVersion: 
+  KeyVaultName: 
+  KubectlVersion: 
+  ResourceGroupLocation: 
+  SubscriptionId: 
+  TemplateSecrets: {}
+
+jobs:
+- deployment: DeployTo_${{ parameters.Environment }}
+  environment: ${{ parameters.Environment }}
+  pool:
+    name: 'DAS - Continuous Integration'
+    ## Linux agent required to ensure the 'kubectl cp' in Copy-FileToContainer.ps1 functions as expected
+    demands: Agent.OS -equals Linux
+  variables:
+  - name: KubernetesNamespace
+    value: monitoring
+  strategy:
+    runOnce:
+      deploy:
+        steps:
+        - checkout: self
+          path: 'das-grafana/'
+        - task: KubectlInstaller@0
+          displayName: 'Install Kubectl ${{ parameters.KubectlVersion }}'
+          inputs:
+            kubectlVersion: ${{ parameters.KubectlVersion }}
+        - task: HelmInstaller@1
+          displayName: 'Install Helm ${{ parameters.HelmClientVersion }}'
+          inputs:
+            helmVersionToInstall: ${{ parameters.HelmClientVersion }}
+        - task: HelmDeploy@0
+          displayName: 'Check application-gateway-ingress deployed'
+          inputs:
+            azureSubscription: '${{ parameters.AzureServiceConnection }}'
+            azureResourceGroup: '${{ parameters.AksResourceGroupName }}'
+            kubernetesCluster: '${{ parameters.AksClusterName }}'
+            useClusterAdmin: true
+            command: get
+            namespace: default
+            arguments: 'all application-gateway-ingress'
+        - template: azure-pipelines-templates/deploy/step/arm-deploy.yml@das-platform-building-blocks
+          parameters:
+            Location: ${{ parameters.ResourceGroupLocation }}
+            ParametersPath: $(Pipeline.Workspace)/das-grafana/azure/template.parameters.json
+            ServiceConnection: ${{ parameters.AzureServiceConnection }}
+            SubscriptionId: ${{ parameters.SubscriptionId }}
+            TemplatePath: $(Pipeline.Workspace)/das-grafana/azure/template.json
+            IsMultiRepoCheckout: true
+            TemplateSecrets: ${{ parameters.TemplateSecrets }}
+        - task: AzurePowerShell@5
+          displayName: 'Azure PowerShell: Set StorageAccountKey variable'
+          inputs:
+            azureSubscription: ${{ parameters.AzureServiceConnection }}
+            scriptType: inlineScript
+            inline: |
+              $Keys = Get-AzStorageAccountKey -ResourceGroupName $(RESOURCEGROUPNAME) -Name $(STORAGEACCOUNTNAME)
+              Write-Output "##vso[task.setvariable variable=StorageAccountKey;isOutput=false]$($Keys[0].Value)"
```

Comment:
Should this be set to secret? e.g. https://github.com/SkillsFundingAgency/das-platform-automation/blob/master/Infrastructure-Scripts/New-StorageAccountSASToken.ps1#L100

Reply:
Implemented